BackendApp/permissionHandler/tests.py

```diff
@@ -0,0 +1,186 @@
+        self.client.force_authenticate(user=self.user)
+        url = reverse_lazy(
+            "shelter_permission_detail", kwargs={"shelter_id": 1, "pk": 1}
+        )
+        response = self.client.get(url)
+        self.assertEqual(
+            response.status_code,
+            status.HTTP_200_OK,
+            f"Expected Response Code 200, received {response.status_code} instead.",
+        )
+
+    def test_edit(self):
+        self.client.force_authenticate(user=self.user)
+        url = reverse_lazy(
+            "shelter_permission_detail", kwargs={"shelter_id": 1, "pk": 1}
+        )
+        data = self.data
+        data["permission"] = 2
+        response = self.client.put(url, data)
+        self.assertEqual(
+            response.status_code,
+            status.HTTP_200_OK,
+            f"Expected Response Code 200, received {response.status_code} instead.",
+        )
+        self.assertEqual(
+            UserPermission.objects.get(id=1).permission.id, data["permission"]
+        )
+
+    def test_remove(self):
+        current_objects_count = UserPermission.objects.count()
+        self.client.force_authenticate(user=self.user)
+        url = reverse_lazy(
+            "shelter_permission_detail", kwargs={"shelter_id": 1, "pk": 1}
+        )
+        response = self.client.delete(url)
+        self.assertEqual(
+            response.status_code,
+            status.HTTP_204_NO_CONTENT,
+            f"Expected Response Code 204, received {response.status_code} instead.",
+        )
+        self.assertEqual(UserPermission.objects.count(), current_objects_count - 1)
+
+
+class PermissionRequestTests(APITestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.data = {"group": "ShelterWorker"}
+        user_data = {
+            "username": "ewa",
+            "password": "ewa12345",
+        }
+        self.user = User.objects.create_user(
+            username=user_data["username"], password=user_data["password"]
+        )
+        self.shelter = Shelter.objects.create()
+        permission_delete = Permission.objects.get(codename="delete_permissionrequest")
+        permission_view = Permission.objects.get(codename="view_permissionrequest")
+        permission_change = Permission.objects.get(codename="change_permissionrequest")
+        UserPermission.objects.create(
+            user=self.user, shelter=self.shelter, permission=permission_delete
+        )
+        UserPermission.objects.create(
+            user=self.user, shelter=self.shelter, permission=permission_view
+        )
+        UserPermission.objects.create(
+            user=self.user, shelter=self.shelter, permission=permission_change
+        )
+        PermissionRequest.objects.create(
+            user=self.user, shelter=self.shelter, group="Manager"
+        )
+
+    def test_list(self):
+        self.client.force_authenticate(user=self.user)
+        url = reverse_lazy("permission_request_list", kwargs={"shelter_id": 1})
```

Comment:
I think with this query we should be getting all permissions requests for all shelters and actually don't allow the user to send more than one request to a given shelter - if he wants to send another one, he must revoke the one he already sent first